# contact state
int8 IN_CONTACT = 1
int8 NO_CONTACT = 0
int8 UNEXPECTED_CONTACT = -1
int8 status